### PR TITLE
[Rendy] inline code documentation pass.

### PR DESCRIPTION
**Pull request by jaynus**
## Description

This is a gentle pass at amethyst_rendy with in-line docs. This is a full documentation inline of amethyst_rendy, allowing it to pass the linter rules:

`#![deny(missing_docs)]`
`#![deny(missing_debug_implementations)]`

## Additions

- Documentation

## Modifications

* This added some missing Debug derives in rendy and assets to resolve that linter.

## PR Checklist

By placing an x in the boxes I certify that I have:

- [x] Ran `cargo test --all` locally if this modified any rs files.
- [x] Ran `cargo +stable fmt --all` locally if this modified any rs files.
- [x] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.

### CODE REVIEW

File: amethyst_rendy/src/batch.rs [hunk 1]
```diff
@@ -1,3 +1,7 @@
+//!  FriziBatch 9000
```
Comment (Frizi) [inline amethyst_rendy/src/batch.rs]: 🙅‍♂

**Comment by jaynus** [comment]
Work complete, ready to merge.

File: amethyst_rendy/src/batch.rs (hunk 1 above)
Comment (jaynus) [reply amethyst_rendy/src/batch.rs]: I'm not changing it.

File: amethyst_rendy/src/lib.rs [hunk 2]
```diff
@@ -1,6 +1,41 @@
-// this is temporary
-// #![allow(dead_code)]
-// #![allow(unused_variables)]
+//! This implementation of the Amethyst Renderer utilizes the `rendy` crate, built on top of
```
Comment (torkleyy) [inline amethyst_rendy/src/lib.rs]: Top-level docs :heart:

File: amethyst_rendy/src/lib.rs [hunk 3]
```diff
@@ -1,6 +1,41 @@
...
+//! * [visibility::BoundingSphere]
+//! * [debug_drawing::DebugLinesComponent]
+//! * [light::Light]
+//! * [resources::Tint]
+//! * [skinning::JointTransforms]
+//! * [sprite::SpriteRender]
+
+#![allow(dead_code)]
+#![allow(unused_variables)]
+#![deny(missing_docs)]
```
Comment (torkleyy) [inline amethyst_rendy/src/lib.rs]: Please lessen to `warn`, we are denying warnings on CI, but don't want to require fixing them when working locally.

File: amethyst_rendy/src/batch.rs (hunk 1 above)
Comment (Frizi) [reply amethyst_rendy/src/batch.rs]: EH.

```suggestion
///! Module containing structures useful for batching draw calls
///! in scenarios with various known assumptions, e.g. order independence.
```

Comment (jaynus) [reply amethyst_rendy/src/batch.rs]: Denied

File: amethyst_rendy/src/formats/texture.rs [hunk 4]
```diff
@@ -15,6 +16,37 @@ use rendy::{
 };
 use serde::{Deserialize, Serialize};
 
+/// Image format description newtype wrapper for `ImageTextureConfig` from rendy.
+///
+/// # Example Usage
+/// ```ignore
```
[the remark this answers is not in the document]
Comment (jaynus) [reply amethyst_rendy/src/formats/texture.rs]: Is there a way to completely straight an entire asset loading pipeline in a doctest? It seemed to cumbersome to do. I'd rather do that later on another documentation pass.

**Comment by torkleyy** [comment]
:tada:
bors r=Frizi,torkleyy

**Pull request merged**